### PR TITLE
added optional chaining in callMethods check

### PR DESCRIPTION
**Pull request by PavelMor25**
## Approach
Added optional chaining in function's arguments "isFunction" which fix check.

## References
[issue 7758](https://github.com/DevExpress/testcafe/issues/7758)

### CODE REVIEW

File: src/client/sandbox/code-instrumentation/methods.ts [hunk 1]
```diff
@@ -77,21 +77,21 @@ export default class MethodCallInstrumentation extends SandboxBase {
                 if (isNullOrUndefined(owner) && !optional)
```
Comment (AlexKamaev) [inline src/client/sandbox/code-instrumentation/methods.ts:77]: I believe that we need to add null reference check here. Smth like follows:
```JS
if (isNullOrUndefined(owner)) {
    if (optional)
        return void 0;

    MethodCallInstrumentation._error(`Cannot call method '${methName}' of ${inaccessibleTypeToStr(owner)}`);
}
```

Comment (AlexKamaev) [reply src/client/sandbox/code-instrumentation/methods.ts:77]: @Aleksey28 please check if proposed solution is appropriate

**Review by AlexKamaev — CHANGES_REQUESTED** [review]

File: src/client/sandbox/code-instrumentation/methods.ts (hunk 1 above)
Comment (Aleksey28) [reply src/client/sandbox/code-instrumentation/methods.ts:77]: Yes, that will be better 👍

File: test/client/fixtures/sandbox/code-instrumentation/process-script-test.js [hunk 2]
```diff
@@ -269,6 +269,14 @@ test('optional chaining', function () {
...
             expected: void 0,
         },
+        {
+            src:      'var obj = null; window.optionChainingResult = obj?.href?.["name"]?.();',
+            expected: void 0,
+        },
+        {
+            src:      'var obj = { href: "123" }; window.optionChainingResult = obj?.link?.["name"]?.();',
+            expected: void 0,
+        },
```
Comment (AlexKamaev) [inline test/client/fixtures/sandbox/code-instrumentation/process-script-test.js:279]: this tests do not work as expected =(
This code does not generates a script with a `__call$` expression. Please research these tests or write new ones to test `__call$` function implementation.

**Review by Aleksey28 — APPROVED** [review]

**Pull request merged**